### ISSUE TITLE
cortex engines init

### ISSUE THREAD

**Issue by vansangpfiev** [janhq/cortex.cpp#47]

**Comment by namchuai** [comment]
@vansangpfiev @nguyenhoangthuan99 , please add if I missing any. Thank you 🙏

**Comment by dan-homebrew** [comment]
@namchuai Can I clarify a few things: 

- CLI command: is it `install` or `init`? If `install`, let's change this issue name
- Will we do an `uninstall` command as well?

**Comment by dan-homebrew** [comment]
@namchuai Another thought is, should the command be:

```
cortex engines install tensorrt-llm
```
or 
```
cortex engines tensorrt-llm install
```

I had originally gone with `cortex engines tensorrt-llm install`, because I initially thought that each of the engines would eventually expose its own CLI commands, e.g: 

```
cortex engines tensorrt-llm settings ...
cortex engines tensorrt-llm gpu 2
```

However, I am increasingly realizing that `install` is an action on `engines`, rather than `tensorrt-llm` (or engine name). In this case, the correct approach should be: 

```
cortex engines install tensorrt-llm
cortex engines uninstall tensorrt-llm
cortex engines list
```

**Comment by namchuai** [comment]
> @namchuai Can I clarify a few things:
> 
> * CLI command: is it `install` or `init`? If `install`, let's change this issue name
> * Will we do an `uninstall` command as well?

1. Yes, it's `install`. I updated the title.
2. I agree about `uninstall`
cc @vansangpfiev @nguyenhoangthuan99 , what do you think?

**Comment by namchuai** [comment]
> @namchuai Another thought is, should the command be:
> 
> ```
> cortex engines install tensorrt-llm
> or 
> cortex engines tensorrt-llm install
> ```
> 
> I had originally gone with `cortex engines tensorrt-llm install`, because I initially thought that each of the engines would eventually expose its own CLI commands, e.g:
> 
> ```
> cortex engines tensorrt-llm settings ...
> cortex engines tensorrt-llm gpu 2
> ```
> 
> However, I realize that `install` is an action on `engines`, rather than `tensorrt-llm` (or engine name). In this case, the correct approach should be:
> 
> ```
> cortex engines install tensorrt-llm
> cortex engines uninstall tensorrt-llm
> cortex engines list
> ```

IMHO, 2nd option is more natural to me.
```
$ cortex engines install tensorrt-llm
```
And I think we should support `options` if any, for example:
```
$ cortex engines install cortex.llamacpp --vulkan 1
```

**Comment by vansangpfiev** [comment]
> > @namchuai Can I clarify a few things:
> > 
> > * CLI command: is it `install` or `init`? If `install`, let's change this issue name
> > * Will we do an `uninstall` command as well?
> 
> 1. Yes, it's `install`. I updated the title.
> 2. I agree about `uninstall`
>    cc @vansangpfiev @nguyenhoangthuan99 , what do you think?

I agree, we should support `uninstall`. @namchuai Can you please create a ticket for it?
